### PR TITLE
new script to calculate wather

### PR DESCRIPTION
**Pull request by mayabrandi**
### Review:
- [x] Code approved by @moonso 
- [x] Tests executed by @mayabrandi 
- [x] "Merge and deploy" approved by @mayabrandi 

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Comment by mayabrandi** [comment]
Testing locally from command line

Skärmavbild 2021-03-17 kl  16 24 40
👍 

Configuring the new script and adding it to the Library Preparation (Cov) step:
Skärmavbild 2021-03-17 kl  16 27 17

Running the script from within the step:
Skärmavbild 2021-03-17 kl  16 30 54
👍

File: cg_lims/EPPs/udf/calculate/base.py [hunk 1]
```diff
@@ -2,10 +2,10 @@
 
 import click
 
-from cg_lims.EPPs.udf.calculate.get_missing_reads import get_missing_reads
 from cg_lims.EPPs.udf.calculate.twist_aliquot_amount import twist_aliquot_amount
 from cg_lims.EPPs.udf.calculate.twist_aliquot_volume import twist_aliquot_volume
 from cg_lims.EPPs.udf.calculate.twist_get_volumes_from_buffer import get_volumes_from_buffer
+from cg_lims.EPPs.udf.calculate.calculate_wather import volume_wather
```
Comment (moonso) [inline cg_lims/EPPs/udf/calculate/base.py:8]: What is `wather`? should it perhaps be `water` or is it another word?

Comment (moonso) [reply cg_lims/EPPs/udf/calculate/base.py:8]: Ok I see now it is water, so perhaps change the spelling?

File: cg_lims/EPPs/udf/calculate/calculate_wather.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    """Calculates volumes of H2O for diluting samples. The total volume differ depending on
+    type of sample. It is given by the process udf 'Total Volume (ul)'."""
+
+    max_volume = process.udf.get("Total Volume (ul)")
+    if not max_volume:
+        raise MissingUDFsError("Process udf missing: Total Volume (ul)")
+
+    missing_udfs = 0
+
+    for art in artifacts:
```
Comment (moonso) [inline cg_lims/EPPs/udf/calculate/calculate_wather.py:24]:
```suggestion
    for artifact in artifacts:
```

File: cg_lims/EPPs/udf/calculate/calculate_wather.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    try:
+        artifacts = get_artifacts(process=process, input=False)
+        calculate_volumes(artifacts=artifacts, process=process)
+        message = "Volume H2O have been calculated for all samples."
+        LOG.info(message)
+        click.echo(message)
+    except LimsError as e:
+        LOG.error(e.message)
+        sys.exit(e.message)
```
Comment (moonso) [inline cg_lims/EPPs/udf/calculate/calculate_wather.py:57]:
```suggestion
        raise click.Abort
```

**Review by moonso — APPROVED** [review]
Looks good, small comments

File: cg_lims/EPPs/udf/calculate/base.py (hunk 1 above)
Comment (mayabrandi) [reply cg_lims/EPPs/udf/calculate/base.py:8]: 👍

File: cg_lims/EPPs/udf/calculate/calculate_wather.py (hunk 2 above)
Comment (mayabrandi) [reply cg_lims/EPPs/udf/calculate/calculate_wather.py:24]: 👍

File: cg_lims/EPPs/udf/calculate/calculate_wather.py (hunk 3 above)
Comment (mayabrandi) [reply cg_lims/EPPs/udf/calculate/calculate_wather.py:57]: Problem is that the message needs to be sent to stderr while exiting, for it to show up as a warning in the info window of a lims step. Ive been strugeling with this before to try to get it working with click.Abort without sucsess. Ill have another look and see if i can find out now...

**Pull request merged**